metricbeat/module/mssql/io/io.go

```diff
@@ -0,0 +1,104 @@
+package io
+
+import (
+	"database/sql"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/pkg/errors"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("mssql", "io", New)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	counter int
+	db      *sql.DB
+}
```

Comment:
The `MetricSet` object is probably going to be reusable between metricsets of this module, take a look for example to the `mongodb` module where there is a common metricset defined at the module level.

Reply:
Done! https://github.com/elastic/beats/pull/8250/commits/fb7cc98b3d9f14b64106fa2b0ad810e17234aad7#diff-eddaad12b395e1f0a6179ce4f2ab9dbfR17